cmd/eksctl-anywhere/cmd/importimages.go

```diff
@@ -83,13 +83,16 @@ func importImages(ctx context.Context, spec string) error {
 	if err != nil {
 		return err
 	}
+	endpoint := net.JoinHostPort(host, port)
+	if port == constants.DefaultHttpsPort {
+		endpoint = host
+	}
 	for _, image := range images {
-		if err := importImage(ctx, de, image.URI, net.JoinHostPort(host, port)); err != nil {
+		if err := importImage(ctx, de, image.URI, endpoint); err != nil {
```

Comment:
how was this tested? It seems like it may have impact across many providers, no?

Reply:
this won't work. 443 is required for some registry (harbor) or it will fail pulling images

Reply:
> > Does appending the ":443" port cause any issues with pushing the images?
> 
> A certain customer registry rejects the push request. They manually tested the push after removing the suffix ":443", and it worked in their environment.

Does the customer set up a proxy in front? Can it be the reason?

Reply:
> > > Does appending the ":443" port cause any issues with pushing the images?
> > 
> > 
> > A certain customer registry rejects the push request. They manually tested the push after removing the suffix ":443", and it worked in their environment.
> 
> Does the customer set up a proxy in front? Can it be the reason?

I don't think they have a proxy in front of the registry. 
They are using 
```
registryMirroConfiguration:
  endpoint: docker.[customer name].com
```

Reply:
yea understand. the conditional check works for me. just it changes the existing eksa behavior a bit (since it proposes to remove the default 443 port). we need to update the doc if we introduce the change.